model/reservation.go

```diff
@@ -0,0 +1,42 @@
+package model
+
+import "time"
+
+// Reservation holds information for a single hotel reservation
+type Reservation struct {
+	ID        string     `json:"id"`
+	Status    string     `json:"status"` // how can I predefine values here?
+	Plan      string     `json:"plan"`   // how can I predefine values here?
+	Adults    int        `json:"adults"`
+	Minors    int        `json:"minors"`
+	AdultFee  float64    `json:"adult_fee"`
+	MinorFee  float64    `json:"minor_fee"`
+	Arrival   *time.Time `json:"arrival"`
+	Departure *time.Time `json:"departure"`
+	CreatedAt *time.Time `json:"created_at"`
+	UpdatedAt *time.Time `json:"updated_at"`
+
+	// Customer information
+	Name  string `json:"name"`
+	Phone string `json:"phone"`
+	Email string `json:"email"`
+}
+
+// totalFee returns the total cost of the reservation
+func (r *Reservation) totalFee() float64 {
+	// where should I validate that the fees are above 0?
```

Comment:
That might be a controller task.

When you receive a new Reservation, make the corresponding checks prior to passing the responsibility to the usecase.